src/abstractdataframe/sort.jl

```diff
@@ -448,8 +476,10 @@ end
 
 Return a permutation vector of row indices of data frame `df` that puts them in
 sorted order according to column(s) `cols`.
+If `cols` were not specified, it returns row indices that sorts `df` lexicographically.
```

Comment:
Rather than repeating this here (which would have to be put in all sorting functions), how about putting `cols=All()` in the signature above? We could also change the actual implementation to make it clearer.

Reply:
This is what I want to do. I will make these changes to this PR and then let @Chandu-4444 to sync the docs following your suggestions and writing that passing no columns leads to an undefined row order.